### PR TITLE
Add ScePower configuration modes

### PR DESCRIPTION
**Pull request by CelesteBlue-dev**

### CODE REVIEW

File: include/psp2/power.h [hunk 1]
```diff
@@ -81,14 +83,19 @@ SceBool scePowerIsBatteryCharging(void);
  */
 int scePowerGetBatteryLifePercent(void);
 
+/*GPU, WLAN/COM configuration setting */
+#define SCE_POWER_CONFIGURATION_MODE_A			0x00000080U /*GPU clock normal, can use WLAN/COM       */
+#define SCE_POWER_CONFIGURATION_MODE_B			0x00000800U /*GPU clock high,   can't use WLAN/COM     */
+#define SCE_POWER_CONFIGURATION_MODE_C			0x00010880U /*GPU clock high,   can use WLAN/COM,      but...   */
```
Comment (devnoname120) [inline include/psp2/power.h]: But?

File: include/psp2/power.h [hunk 2]
```diff
@@ -81,14 +83,19 @@ SceBool scePowerIsBatteryCharging(void);
  */
 int scePowerGetBatteryLifePercent(void);
 
+/*GPU, WLAN/COM configuration setting */
+#define SCE_POWER_CONFIGURATION_MODE_A			0x00000080U /*GPU clock normal, can use WLAN/COM       */
```
Comment (devnoname120) [inline include/psp2/power.h]: Please use an `enum` instead.

File: include/psp2/power.h [hunk 3]
```diff
@@ -81,14 +83,19 @@ SceBool scePowerIsBatteryCharging(void);
...
+
 /**
  * Set configuration mode ?
  *
  * @param mode - The mode to set
  *
  * @return ?
  */
-int scePowerSetConfigurationMode(int mode);
+int scePowerSetConfigurationMode(int conf);
```
Comment (devnoname120) [inline include/psp2/power.h]: Change the documentation as well, and add `::ScePowerConfigurationMode`.

File: include/psp2/power.h (hunk 1 above)
Comment (CelesteBlue-dev) [reply include/psp2/power.h]: but uses battery more intensively

File: include/psp2/power.h (hunk 2 above)
Comment (CelesteBlue-dev) [reply include/psp2/power.h]: why would an enum better ?

File: include/psp2/power.h (hunk 3 above)
Comment (CelesteBlue-dev) [reply include/psp2/power.h]: where is the documention ?

File: include/psp2/power.h [hunk 4]
```diff
@@ -36,9 +36,11 @@ typedef enum ScePowerCallbackType {
 	SCE_POWER_CB_RESUMING         = 0x00020000,
 	/** indicates the unit is suspending, seems to occur due to inactivity */
 	SCE_POWER_CB_SUSPENDING       = 0x00010000,
-	/**indicates the unit is plugged into an AC outlet*/
+	/** indicates the unit is plugged into an AC outlet */
 	SCE_POWER_CB_AC_POWER         = 0x00001000,
-	/**indicates there is a battery present in the unit**/
+	/** indicates the battery is in low state **/
+	SCE_POWER_CB_LOWBATTERY         = 0x00000100,
```
Comment (Rinnegatamante) [inline include/psp2/power.h]: Not properly idented

File: include/psp2kern/power.h [hunk 5]
```diff
@@ -25,9 +25,11 @@ typedef enum ScePowerCallbackType {
 	SCE_POWER_CB_RESUMING         = 0x00020000,
 	/** indicates the unit is suspending, seems to occur due to inactivity */
 	SCE_POWER_CB_SUSPENDING       = 0x00010000,
-	/**indicates the unit is plugged into an AC outlet*/
+	/** indicates the unit is plugged into an AC outlet */
 	SCE_POWER_CB_AC_POWER         = 0x00001000,
-	/**indicates there is a battery present in the unit**/
+	/** indicates the battery is in low state **/
+	SCE_POWER_CB_LOWBATTERY         = 0x00000100,
```
Comment (Rinnegatamante) [inline include/psp2kern/power.h]: Not properly idented

**Comment by CelesteBlue-dev** [comment]
@devnoname120 I don't know if I added ::ScePowerConfigurationMode at the good location
I have fixed all the requested changes.

File: include/psp2/power.h [hunk 6]
```diff
@@ -82,13 +91,17 @@ SceBool scePowerIsBatteryCharging(void);
...
 /**
- * Set configuration mode ?
+ * Set power configuration mode between:
+ *
+ * Mode A - This is the normal mode at process start-up. The clock frequency of the GPU core is the "normal" clock frequency. The WLAN/COM can be used.
+ * Mode B - This mode accelerates the GPU clock frequency. The clock frequency of the GPU core is the "high" clock frequency. The WLAN/COM cannot be used.
+ * Mode C - This mode accelerates the GPU clock frequency, and also uses the WLAN/COM. The clock frequency of the GPU core is the "high" clock frequency, and use of the WLAN/COM is possible. The screen (touchscreen) brightness, however, is limited. Also, camera cannot be used.
  *
- * @param mode - The mode to set
+ * @param ::ScePowerConfigurationMode conf - The mode to set
```
Comment (devnoname120) [inline include/psp2/power.h]:
```
@param conf One of ::ScePowerConfigurationMode
```

File: include/psp2/power.h [hunk 7]
```diff
@@ -82,13 +91,17 @@ SceBool scePowerIsBatteryCharging(void);
...
  *
- * @return ?
+ * @return 0 on success
  */
-int scePowerSetConfigurationMode(int mode);
+int scePowerSetConfigurationMode(ScePowerConfigurationMode conf);
```
Comment (devnoname120) [inline include/psp2/power.h]: Passing an enum as an argument gives us no guarantee regarding its size. Please use `int` instead.

File: include/psp2/power.h [hunk 8]
```diff
@@ -36,12 +36,21 @@ typedef enum ScePowerCallbackType {
...
-	/**indicates there is a battery present in the unit**/
+	/** indicates the battery is in low state **/
+	SCE_POWER_CB_LOWBATTERY       = 0x00000100,
+	/** indicates there is a battery present in the unit **/
 	SCE_POWER_CB_BATTERY_EXIST    = 0x00000080
 } ScePowerCallbackType;
 
+/*GPU, WLAN/COM configuration setting */
+typedef enum ScePowerConfigurationMode {
+	SCE_POWER_CONFIGURATION_MODE_A   = 0x00000080U, /*GPU clock normal, can use WLAN/COM       */
```
Comment (devnoname120) [inline include/psp2/power.h]: Extra whitespace before the `*/`.

**Comment by CelesteBlue-dev** [comment]
done. thanks for all @devnoname120 and Rinne.

**Pull request merged**